### PR TITLE
feat: Add SwiftDescriptor error fix to troubleshooting for RN and Apple

### PR DESCRIPTION
**Pull request by krystofwoldrich**
Based on the reports from `sentry-cocoa` and `sentry-react-native`, we should add this to our docs.

- https://github.com/getsentry/sentry-react-native/issues/3765
- https://github.com/getsentry/sentry-cocoa/issues/3850

### CODE REVIEW

**Review by brustolin — APPROVED** [review]

File: docs/platforms/apple/common/troubleshooting/index.mdx [hunk 1]
```diff
@@ -50,6 +50,24 @@ Check your app for any procedure that cleans the cache of your app during runtim
 
 Since Xcode 14 App Store Connect doesn't make debug symbols available for download anymore, see [Xcode Release Notes](https://developer.apple.com/documentation/xcode-release-notes/xcode-14-release-notes#Deprecations). Please use a different way of uploading the debug symbols by following the [documentation](/platforms/apple/dsym/).
 
+## Unknown receiver 'SwiftDescriptor'
+
+If you see an error like `Unknown receiver 'SwiftDescriptor'` during the application build process, it's due to a conflict of `APPLICATION_EXTENSION_API_ONLY`. The Sentry cocoapod is by default configured with `APPLICATION_EXTENSION_API_ONLY=YES`, but some `post_install` script changed it. To fix the error add the following to the `Podfile`.
```
Comment (philipphofmann) [inline docs/platforms/apple/common/troubleshooting/index.mdx:55]: `l`: Minor fixes
```suggestion
If you see an error like `Unknown receiver 'SwiftDescriptor'` during the application build process, it's due to a conflict of `APPLICATION_EXTENSION_API_ONLY`. By default, the Sentry CocoaPod is configured with `APPLICATION_EXTENSION_API_ONLY=YES` but some `post_install` script changed it. To fix the error, add the following to the `Podfile`.
```

**Review by philipphofmann — APPROVED** [review]
LGTM

File: docs/platforms/apple/common/troubleshooting/index.mdx (hunk 1 above)
Comment (vivianyentran) [reply docs/platforms/apple/common/troubleshooting/index.mdx:55]:
```suggestion
If you see an error like `Unknown receiver 'SwiftDescriptor'` during the application build process, there's a conflict stemming from `APPLICATION_EXTENSION_API_ONLY`. By default, the Sentry Cocoapod is configured with `APPLICATION_EXTENSION_API_ONLY=YES`, but some `post_install` script likely changed it. To fix this error, add the following to the `Podfile`.
```

File: docs/platforms/react-native/troubleshooting/index.mdx [hunk 2]
```diff
@@ -186,6 +186,30 @@ The Xcode integration reads `dist` from `CFBundleVersion` but Xcode Cloud builds
 export SENTRY_DIST=$CI_BUILD_NUMBER
 ```
 
+## Unknown receiver 'SwiftDescriptor'
+
+If you see an error like `Unknown receiver 'SwiftDescriptor'` during the application build process, it's due to a conflict of `APPLICATION_EXTENSION_API_ONLY`. The Sentry cocoapod is by default configured with `APPLICATION_EXTENSION_API_ONLY=YES`, but some `post_install` script changed it. To fix the error add the following to the `Podfile`.
```
Comment (vivianyentran) [inline docs/platforms/react-native/troubleshooting/index.mdx:191]:
```suggestion
If you see an error like `Unknown receiver 'SwiftDescriptor'` during the application build process, there's a conflict stemming from `APPLICATION_EXTENSION_API_ONLY`. By default, the Sentry Cocoapod is configured with `APPLICATION_EXTENSION_API_ONLY=YES`, but some `post_install` script likely changed it. To fix this error, add the following to the `Podfile`.```

**Review by vivianyentran — APPROVED** [review]
small edit, but LGTM!

**Comment by philipphofmann** [comment]
@krystofwoldrich, do you maybe want to reopen this and merge it?

**Review by philipphofmann — APPROVED** [review]
LGTM

**Comment by krystofwoldrich** [comment]
@philipphofmann @kahest Thank you for publishing this, I've missed that it got closed.

**Pull request merged**